### PR TITLE
Fix Florence2 training-loss double-shift (same pattern as Moonshine #…

### LINKED ISSUE #46897: Florence2 training-loss double-shift bug (same pattern as Moonshine #46784)
### System Info

- `transformers` version: 5.13.0.dev0 (main branch at commit dc5a497c18)
- - Platform: Windows
- - - Python version: 3.x
- - - - Using GPU: N/A (code-level bug, not runtime)

### Who can help?

@zucchini-nlp @yonigozlan

### Information

- [x] The official example scripts
- [ ] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

aIn `src/transformers/models/florence2/modular_florence2.py`, `Florence2ForConditionalGeneration.forward()` has a double-shift bug introduced in PR #40914:

1. Labels are shifted right via `shift_tokens_right()` (line 1588) to create `decoder_input_ids`
2. 2. Then `self.loss_function()` is called (line 1613), which maps to `ForCausalLMLoss`
3. 3. `ForCausalLMLoss` shifts labels again internally (`labels[..., 1:]` in `loss/loss_utils.py` line 63-64)
This means the model trains against `labels[..., 1:]` instead of `labels`. Same bug fixed in Moonshine PR #46784.

```python
# Buggy code:
loss = self.loss_function(logits=logits, labels=labels, vocab_size=self.config.text_config.vocab_size, **kwargs)

# Should be:
loss_fct = CrossEntropyLoss()
loss = loss_fct(logits.reshape(-1, self.config.text_config.vocab_size), labels.reshape(-1))
```

### Expected behavior

Training loss should be computed directly against `labels` without any additional shifting, since `shift_tokens_right()` already handles the decoder input alignment. This matches the behavior of Whisper, Bart, and the now-fixed Moonshine and CohereASR models. The fix should use `CrossEntropyLoss()` instead of `self.loss_function()`, as done in Moonshine PR #46784.

### PR DESCRIPTION
**Pull request by sharmax-vikas**
[CI](https://transformers-ci.lor-e.huggingface.cool/d/pytest-observability-by-pr/pytest-observability-branch?var-pr=46898)

# What does this PR do?

Fixes #46897

Fix double-shift training-loss bug in `Florence2ForConditionalGeneration` - the exact same pattern fixed in Moonshine (#46784) and CohereASR (#46894).

`Florence2ForConditionalGeneration.forward()` shifts labels right via `shift_tokens_right()` to create `decoder_input_ids`, then calls `self.loss_function()` which maps to `ForCausalLMLoss`. That function shifts labels again internally (`labels[..., 1:]`), so the model trains against `labels[..., 1:]` instead of `labels`.

The fix replaces `self.loss_function()` with plain `CrossEntropyLoss()`, matching the approach used in Whisper, Bart, and the now-fixed Moonshine.

Both `modular_florence2.py` (source of truth) and `modeling_florence2.py` (auto-generated) are updated.

## Who can review?

@zucchini-nlp @yonigozlan (multimodal/vision-language models)

---

### CI recap

**Dashboard:** [View test results in Grafana](https://transformers-ci.lor-e.huggingface.cool/d/pytest-observability-by-pr/pytest-observability-branch?var-pr=46898)
**Latest run:** [28365382183:2](https://github.com/huggingface/transformers/actions/runs/28365382183)
**Result:** `success` | **Jobs:** 2 | **Tests:** 19 | **Failures:** 0 | **Duration:** 2m 2s

### CODE REVIEW

File: src/transformers/models/florence2/modeling_florence2.py [hunk 1]
```diff
@@ -902,9 +903,8 @@ def forward(
 
         loss = None
         if labels is not None:
-            loss = self.loss_function(
-                logits=logits, labels=labels, vocab_size=self.config.text_config.vocab_size, **kwargs
-            )
+            loss_fct = CrossEntropyLoss()
```
Comment (zucchini-nlp) [inline src/transformers/models/florence2/modeling_florence2.py]: we can pass `shift_labels=labels` into loss-fn to not sift it twice

File: src/transformers/models/florence2/modeling_florence2.py [hunk 2]
```diff
@@ -872,6 +872,7 @@ def forward(
         >>> processor.batch_decode(generate_ids, skip_special_tokens=True, clean_up_tokenization_spaces=False)[0]
         "A green car parked in front of a yellow building."
         ```"""
+        do_shift_labels = False
```
Comment (zucchini-nlp) [inline src/transformers/models/florence2/modeling_florence2.py]: `do_shift_labels` is always `True` imo since users prepare labels by copying input ids, and shift decoder-input-ids when created by a collator. CMIIW, I can't remember a train use-case with non-shifted inputs

Comment (sharmax-vikas) [reply src/transformers/models/florence2/modeling_florence2.py]: Simplified - removed the `do_shift_labels` flag entirely, now always passing `shift_labels=labels` when labels are present. Thanks for the feedback!

**Review by zucchini-nlp — APPROVED** [review]
Thanks, merging
